file_store/src/file_upload.rs

```diff
@@ -75,14 +67,13 @@ impl ManagedTask for FileUploadServer {
 
 impl FileUploadServer {
     pub async fn run(self, shutdown: triggered::Listener) -> Result {
-        tracing::info!("starting file uploader {}", self.store.bucket);
+        tracing::info!("starting file uploader {}", self.bucket);
 
         let uploads = self
             .messages
-            .map(|msg| (self.store.clone(), msg))
-            .for_each_concurrent(5, |(store, path)| async move {
+            .map(|msg| (self.client.clone(), self.bucket.clone(), msg))
+            .for_each_concurrent(5, |(client, bucket, path)| async move {
```

Comment:
Potential for future updates.

This function takes `self`, we can grab references to instead of cloning.

```rs
let client = &self.client;
let bucket = &self.bucket;
```

That would allow us to remove the `.map()` call.